plugins/modules/proxysql_cluster.py

```diff
@@ -0,0 +1,443 @@
+- community.proxysql.proxysql.managing_config
+- community.proxysql.proxysql.connectivity
+
+'''
+
+EXAMPLES = '''
+---
+# This example adds a server, it saves the proxysql server config to disk, but
+# avoids loading the proxysql server config to runtime (this might be because
+# several servers are being added and the user wants to push the config to
+# runtime in a single batch using the community.general.proxysql_manage_config
+# module).  It uses supplied credentials to connect to the proxysql admin
+# interface.
+
+- name: Add a server
+  community.proxysql.proxysql_cluster:
+    login_user: 'admin'
+    login_password: 'admin'
+    hostname: 'proxysql-01'
+    port: 6032
+    weight: 1
+    comment: 'Main ProxySQL server'
+    state: present
+    save_to_disk: True
+    load_to_runtime: False
+
+# This example removes a server, saves the proxysql server config to disk, and
+# dynamically loads the proxysql server config to runtime.  It uses credentials
+# in a supplied config file to connect to the proxysql admin interface.
+
+- name: Remove a server
+  community.proxysql.proxysql_cluster:
+    config_file: '~/proxysql.cnf'
+    hostname: 'proxysql-01'
+    save_to_disk: True
+    state: absent
+'''
+
+RETURN = '''
+stdout:
+    description: The proxysql host modified or removed from ProxySQL cluster
+    returned: On create/update will return the newly modified host, on delete
+              it will return the deleted record.
+    type: dict
+    "sample": {
+        "changed": true,
+        "hostname": "10.10.10.2",
+        "msg": "Added server to proxysql cluster",
+        "server": {
+            "hostname": "10.10.10.2",
+            "port": "6032",
+            "weight": "1",
+            "comment": ""
+        },
+        "state": "present"
+    }
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.community.proxysql.plugins.module_utils.mysql import mysql_connect, mysql_driver, mysql_driver_fail_msg
+from ansible.module_utils.six import iteritems
+from ansible.module_utils._text import to_native
+
+# ===========================================
+# proxysql module specific support methods.
+#
+
+
+def perform_checks(module):
+    if module.params["login_port"] < 0 \
+       or module.params["login_port"] > 65535:
+        module.fail_json(
+            msg="login_port must be a valid unix port number (0-65535)"
+        )
+
+    if module.params["port"] < 0 \
+       or module.params["port"] > 65535:
+        module.fail_json(
+            msg="port must be a valid unix port number (0-65535)"
+        )
+
+    if mysql_driver is None:
+        module.fail_json(msg=mysql_driver_fail_msg)
+
+
+def save_config_to_disk(cursor):
+    cursor.execute("SAVE PROXYSQL SERVERS TO DISK")
+    return True
+
+
+def load_config_to_runtime(cursor):
+    cursor.execute("LOAD PROXYSQL SERVERS TO RUNTIME")
+    return True
+
+
+class ProxySQLServer(object):
+
+    def __init__(self, module):
```

Comment:
```suggestion
    def __init__(self, module):
```
Most the things from the other PR is applicable here.